src/main/scala/xiangshan/backend/fu/CSR.scala

```diff
@@ -1258,99 +1239,47 @@ class PFEvent(implicit p: Parameters) extends XSModule with HasCSRConst  {
-  val csrevent4  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent5  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent6  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent7  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent8  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent9  = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent10 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent11 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent12 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent13 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent14 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent15 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent16 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent17 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent18 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent19 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent20 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent21 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent22 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent23 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent24 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent25 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent26 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent27 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent28 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent29 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent30 = RegInit(UInt(XLEN.W), 0.U)
-  val csrevent31 = RegInit(UInt(XLEN.W), 0.U)
+  val perfEvents = List.fill(8)(RegInit("h0000000000".U(XLEN.W))) ++ 
+                   List.fill(8)(RegInit("h4010040100".U(XLEN.W))) ++ 
+                   List.fill(8)(RegInit("h8020080200".U(XLEN.W))) ++ 
+                   List.fill(5)(RegInit("hc0300c0300".U(XLEN.W)))
 
   var perfEventMapping = Map(
-    MaskedRegMap(Mhpmevent3, csrevent3 ),
-    MaskedRegMap(Mhpmevent4, csrevent4 ),
-    MaskedRegMap(Mhpmevent5, csrevent5 ),
-    MaskedRegMap(Mhpmevent6, csrevent6 ),
-    MaskedRegMap(Mhpmevent7, csrevent7 ),
-    MaskedRegMap(Mhpmevent8, csrevent8 ),
-    MaskedRegMap(Mhpmevent9, csrevent9 ),
-    MaskedRegMap(Mhpmevent10,csrevent10),
-    MaskedRegMap(Mhpmevent11,csrevent11),
-    MaskedRegMap(Mhpmevent12,csrevent12),
-    MaskedRegMap(Mhpmevent13,csrevent13),
-    MaskedRegMap(Mhpmevent14,csrevent14),
-    MaskedRegMap(Mhpmevent15,csrevent15),
-    MaskedRegMap(Mhpmevent16,csrevent16),
-    MaskedRegMap(Mhpmevent17,csrevent17),
-    MaskedRegMap(Mhpmevent18,csrevent18),
-    MaskedRegMap(Mhpmevent19,csrevent19),
-    MaskedRegMap(Mhpmevent20,csrevent20),
-    MaskedRegMap(Mhpmevent21,csrevent21),
-    MaskedRegMap(Mhpmevent22,csrevent22),
-    MaskedRegMap(Mhpmevent23,csrevent23),
-    MaskedRegMap(Mhpmevent24,csrevent24),
-    MaskedRegMap(Mhpmevent25,csrevent25),
-    MaskedRegMap(Mhpmevent26,csrevent26),
-    MaskedRegMap(Mhpmevent27,csrevent27),
-    MaskedRegMap(Mhpmevent28,csrevent28),
-    MaskedRegMap(Mhpmevent29,csrevent29),
-    MaskedRegMap(Mhpmevent30,csrevent30),
-    MaskedRegMap(Mhpmevent31,csrevent31),
+    MaskedRegMap(Mhpmevent3 , perfEvents( 0), "hf87fff3fcff3fcff".U(XLEN.W)),
```

Comment:
Maybe the Mhpmevent*s and Mhpmcounter*s could be lists.
If the declarations in CSRConsts.scala are expected to be kept, we could define another list such as 'Mhpmevents', and let them correspond with the previous decls